File: 3074-apple-redistribution-into-boxes/3074-apple-redistribution-into-boxes.py
class Solution:
    def minimumBoxes(self, apple: List[int], c: List[int]) -> int:
        x = sum(apple)
        v = 0
        c.sort(reverse = True)
        n = len(c)
        for i in range(n):
            v += c[i]
            if v >= x:
                return i + 1
        return -1

File: 3074-apple-redistribution-into-boxes/3074-apple-redistribution-into-boxes.cpp
class Solution {
public:
    int minimumBoxes(vector<int>& apple, vector<int>& cap) {
        int n = size(apple);
        int m = size(cap);
        int v = 0, x = accumulate(begin(apple), end(apple), 0);
        sort(rbegin(cap), rend(cap));
        //for(auto& c : cap) cout<<c<<" ";
        for (int i = 0; i < m; i++) {
            v += cap[i];
            if (v >= x)
                return i + 1;
        }
        return -1;
    }
};